### PR TITLE
Feature/netcore1.0

### PR DESCRIPTION
**Pull request by alyiox**
1. Migration from DNX RC1 to NET Core CLI.
2. Target framework to `netstandard1.0`
3. Support to custom cache policy by `CacheEntryOptions`
4. Update version from `1.2.0-beta1` to `1.2.1`

### CODE REVIEW

File: src/DotQuery.Extensions/MemoryCacheBasedQueryCache.cs [hunk 1]
```diff
@@ -1,43 +1,56 @@
...
         }
 
         public void Clear()
         {
-            m_objectCache.Trim(100);
+            _memoryCache.Compact(100);
         }
 
-        public TValue GetOrAdd(TKey key, TValue lazyTask)
+        public TValue GetOrAdd(TKey key, TValue lazyTask) => GetOrAdd(key, lazyTask, new CacheEntryOptions { SlidingExpiration = _slidingExpiration });
```
Comment (karldodd) [inline src/DotQuery.Extensions/MemoryCacheBasedQueryCache.cs]: Avoid using 'new CacheEntryOptions' if we can reuse an instance

File: src/DotQuery.Extensions/project.json [hunk 2]
```diff
@@ -1,32 +1,29 @@
 {
-  "version": "1.2.0-beta1",
+  "version": "1.2.1",
```
Comment (karldodd) [inline src/DotQuery.Extensions/project.json]: Let's make it to 1.3 as upgrading to .net core 1.0 is a huge step forward.

File: src/DotQuery.Core/AsyncQueryExecutor.cs [hunk 3]
```diff
@@ -42,6 +42,23 @@ public Task<TResult> QueryAsync(TQuery query)
         }
 
         /// <summary>
+        /// Execute the given query as an asynchronous operation with a specified cache policy.
+        /// </summary>
+        /// <param name="query">The query to be executed</param>
+        /// <param name="options">The cache policy options.</param>
+        /// <returns>The task object representing the asynchronous operation</returns>
+        public Task<TResult> QueryAsync(TQuery query, CacheEntryOptions options)
```
Comment (karldodd) [inline src/DotQuery.Core/AsyncQueryExecutor.cs]: Any reason we need a standalone `CacheEntryOptions`? Would be great if we integrate cache entry option into exsiting QueryOptions class (as a field like CacheOption). @qqbuby

Comment (alyiox) [inline src/DotQuery.Core/AsyncQueryExecutor.cs]: It's a pity that `QueryOptions` is a flags `enum` type, not a `class`. So create a new method signature with `QueryAsync(TQuery, CacheEntryOptions)` for the  the backward compatibility.

File: src/DotQuery.Extensions/MemoryCacheBasedQueryCache.cs (hunk 1 above)
Comment (alyiox) [inline src/DotQuery.Extensions/MemoryCacheBasedQueryCache.cs]: fixed

File: src/DotQuery.Extensions/project.json (hunk 2 above)
Comment (alyiox) [inline src/DotQuery.Extensions/project.json]: fixed

File: src/DotQuery.Core/AsyncQueryExecutor.cs (hunk 3 above)
Comment (karldodd) [inline src/DotQuery.Core/AsyncQueryExecutor.cs]: @qqbuby Try merge QueryOptions and CacheEntryOptions to a single unified class

**Comment by karldodd** [comment]
Will merge and release an beta version @qqbuby

**Pull request merged**